pkg/contracts/ton/gateway_test.go

```diff
@@ -0,0 +1,313 @@
+package ton
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tonkeeper/tongo/boc"
+	"github.com/tonkeeper/tongo/tlb"
+	"github.com/tonkeeper/tongo/ton"
+)
+
+func TestParsing(t *testing.T) {
+	swapBodyAndParse := func(gw *Gateway, tx ton.Transaction, body *boc.Cell) *Transaction {
+		tx.Msgs.InMsg.Value.Value.Body.Value = tlb.Any(*body)
+
+		parsed, err := gw.ParseTransaction(tx)
+		require.NoError(t, err)
+
+		return parsed
+	}
+
+	t.Run("Donate", func(t *testing.T) {
+		// ARRANGE
+		// Given a tx
+		tx, fx := getFixtureTX(t, "00-donation")
+
+		// Given a gateway contract
+		gw := NewGateway(ton.MustParseAccountID(fx.Account))
+
+		// ACT
+		parsedTX, err := gw.ParseTransaction(tx)
+
+		// ASSERT
+		require.NoError(t, err)
+
+		assert.Equal(t, int(OpDonate), int(parsedTX.Operation))
+		assert.Equal(t, true, parsedTX.IsInbound())
+
+		const (
+			expectedSender   = "0:9594c719ec4c95f66683b2fb1ca0b09de4a41f6fb087ba4c8d265b96a4cce50f"
+			expectedDonation = 1_499_432_947 // 1.49... TON
+		)
+
+		donation, err := parsedTX.Donation()
+		assert.NoError(t, err)
+		assert.Equal(t, expectedSender, donation.Sender.ToRaw())
+		assert.Equal(t, expectedDonation, int(donation.Amount.Uint64()))
+
+		// Check that AsBody works
+		var (
+			parsedTX2 = swapBodyAndParse(gw, tx, lo.Must(donation.AsBody()))
+			donation2 = lo.Must(parsedTX2.Donation())
+		)
+
+		assert.Equal(t, donation, donation2)
+	})
+
+	t.Run("Deposit", func(t *testing.T) {
+		// ARRANGE
+		// Given a tx
+		tx, fx := getFixtureTX(t, "01-deposit")
```

Comment:
Is this tx in TONScan?

Reply:
yes, you can check json fixtures. They have "description" link; it contains link to the tx